### ISSUE TITLE
How to structure a simple page from components?

### ISSUE THREAD

**Issue by atomiks** [hyperapp/hyperapp#650]
Hi, I've started experimenting with hyperapp but still trying to grasp the fundamentals.

Does it make sense to make a component out of something that only has a single instance?

For example in the following, `Counter` only has a single instance in the view, and it's nested, so the state and actions seem to need to be passed down which seems antipatternish. I tried to use the example from "lazy components", but wasn't able to get it working.

Basically, how should an app consisting of a navbar, counter and footer be structured? Like, if a page has 10 individual areas that aren't reused, I'd generally make them a component for neatness' sake in the view so each section can be a single line in the main view function.

```jsx
import { h, app } from 'hyperapp'

const state = {
  count: 0
}

const actions = {
  up: value => state => ({ count: state.count + value }),
  down: value => state => ({ count: state.count - value }),
  reset: () => ({ count: 0 })
}

// Multiple instances, makes sense to be a component
const CounterButton = ({ text, action, by }) => (
  <button class="CounterButton" onclick={() => action(by)}>
    {text}
  </button>
)

 // Single instance...
const Counter = ({ state, actions }) => (
  <div class="Counter">
    <h1 class="Counter__value">{state.count}</h1>
    <CounterButton text="-" action={actions.down} by={1} />
    <CounterButton text="reset" action={actions.reset} />
    <CounterButton text="+" action={actions.up} by={1} />
  </div>
)

const view = (state, actions) => (
  <main class="container">
    ... navbar ...
    <Counter state={state} actions={actions} />
    ... footer ...
  </main>
)

app(state, actions, view, document.body)
```

**Comment by SkaterDad** [comment]
> For example in the following, Counter only has a single instance in the view, and it's nested, so the state and actions seem to need to be passed down which seems antipatternish. I tried to use the example from "lazy components", but wasn't able to get it working.

Lazy Components are a new feature that is not yet released.  The documentation is for the `master` branch of the project.   @jorgebucaran said in another Issue that he plans on releasing soon (possibly today).   Once that lands, you can use the new feature to remove the need for passing down the global state & actions.

```jsx
 // Single instance...
const Counter = () => (state, actions) => (
  <div class="Counter">
    <h1 class="Counter__value">{state.count}</h1>
    <CounterButton text="-" action={actions.down} by={1} />
    <CounterButton text="reset" action={actions.reset} />
    <CounterButton text="+" action={actions.up} by={1} />
  </div>
)

const view = (state, actions) => (
  <main class="container">
    ... navbar ...
    <Counter  />
    ... footer ...
  </main>
)

```

**Comment by atomiks** [comment]
Think this can be closed now with the introduction of lazy components a while ago
